sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,217 @@ class TableCheckpointStore:
+        return checkpointslist
+
+    def update_checkpoint(self, checkpoint):
+        """Updates the checkpoint using the given information for the offset, associated partition and
+        consumer group in the storage table.
+        Note: If you plan to implement a custom checkpoint store with the intention of running between
+        cross-language EventHubs SDKs, it is recommended to persist the offset value as an string.
+        :param Dict[str,Any] checkpoint: A dict containing checkpoint information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the checkpoint is associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `sequence_number` (int): The sequence number of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+                - `offset` (str): The offset of the :class:`EventData<azure.eventhub.EventData>`
+                  the new checkpoint will be associated with.
+        :rtype: None
+        """
+        try:
+            checkpoint_entity = self._create_checkpoint_entity(checkpoint)
+            self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=checkpoint_entity)
+        except ResourceNotFoundError:
+            self._checkpoint_entity(checkpoint)
+
+    def _update_ownership(self, ownership):
+        try:
+            if ownership['etag'] is None:
+                raise ResourceNotFoundError
+            ownership_entity = self._create_ownership_entity(ownership)
+            metadata = self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=ownership_entity,
+            etag=ownership['etag'], match_condition=MatchConditions.IfNotModified)
+            ownership['etag'] = metadata['etag']
+            ownership['lastmodifiedtime'] = metadata['date']
+        except (ResourceNotFoundError, ValueError):
+            try:
+                metadata = self._ownership_entity(ownership)
+                ownership['etag'] = metadata['etag']
+                ownership['lastmodifiedtime'] = metadata['date']
+            except ResourceExistsError:
+                raise OwnershipLostError()
+        except ResourceModifiedError:
+            raise OwnershipLostError()
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
 
-    def update_checkpoint(self, checkpoint, **kwargs):
-        pass
+    def _claim_one_partition(self, ownership):
+        self._update_ownership(ownership)
+        return ownership
```

Comment:
This method is basically 1 line, so I don't think it is needed. You can just call `self._update_ownership` directly instead of `_claim_one_partition`.

Reply:
okay I removed _cliam_one_partition